Adding a --tilejson arg to the show command

mikeocool opened the pull request:
And arg to the show command that will output the full tile json -- reusing the code that generates the tilejson from the serve command: `pmtiles show --tilejson test.pmtiles`

One part of this that's currently lacking is specifying the URL used in`tiles` -- currently it just outputs `/{z}/{x}/{y}.mvt`.  Was thinking either:
* add an arg to the show command that allows specifying the tiles URL -- doesn't seem totally ideal, since that arg won't do anything unless you are also using the tilejson arg.
* Maybe breakout the tilejson into it's own command and add the arg to that, so it'd be called like: `pmtiles tilejson --tiles-url http://example.com/tiles.pmtiles/ tiles.pmtiles`

bdon commented:
I guess I imagined the main use case here is doing something like:

`pmtiles show --tilejson foo.pmtiles > foo.json` 

The TileJSON spec unfortunately requires absolute and not relative URLs for the tiles endpoint. Options I see are:

* add a second flag called `--public-hostname` just like with `pmtiles serve`
* output an obvious placeholder like `https://example.com/{z}/{x}/{y}.mvt` and a warning on `stderr` if the public hostname is not passed

I think it's ok to have that flag be there and only be relevant for `--tilejson`, I think it's cleaner than making a whole new subcommand. 

Is the context you will use this different from the above?

mikeocool commented:
@bdon Yeah, that's pretty much exactly the use case I had in mind.

Went ahead and added a `--public-url` arg -- that allows you to specify the FQDN for the tiles -- in case you aren't just hosting the tiles at `https://whatever.com/key/{z}/{x}/{y}.mvt` -- and added the warning and the placeholder if you don't specify that arg.

So you should now be able to do: `pmtiles show --tilejson test.pmtiles` or `pmtiles show --tilejson --public-url https://example.com/test.pmtiles/{z}/{x}/{y} test.pmtiles`

bdon commented:
Thanks! I have some small additions I'll add in a followup PR
